common/djangoapps/student/tests/test_email.py

```diff
@@ -233,78 +236,42 @@ def assertFailedRequest(self, response_data, expected_error):
-        self.assertFailedRequest(self.run_request(), 'Invalid password')
-
     @patch('student.views.render_to_string', Mock(side_effect=mock_render_to_string, autospec=True))
     def test_duplicate_activation_key(self):
-        """Assert that if two users change Email address simultaneously, server should return 200"""
+        """Assert that if two users change Email address simultaneously, no error is thrown"""
 
         # New emails for the users
         user1_new_email = "valid_user1_email@example.com"
         user2_new_email = "valid_user2_email@example.com"
 
-        # Set new email for user1.
-        self.request.POST['new_email'] = user1_new_email
-
         # Create a another user 'user2' & make request for change email
         user2 = UserFactory.create(email=self.new_email, password="test2")
-        user2_request = self.req_factory.post('unused_url', data={
-            'password': 'test2',
-            'new_email': user2_new_email
-        })
-        user2_request.user = user2
 
-        # Send requests & check if response was successful
-        user1_response = change_email_request(self.request)
-        user2_response = change_email_request(user2_request)
-
-        self.assertEqual(user1_response.status_code, 200)
-        self.assertEqual(user2_response.status_code, 200)
+        # Send requests & ensure no error was thrown
+        self.assertIsNone(self.do_email_change(self.user, user1_new_email))
+        self.assertIsNone(self.do_email_change(user2, user2_new_email))
 
     def test_invalid_emails(self):
         for email in ('bad_email', 'bad_email@', '@bad_email'):
-            self.request.POST['new_email'] = email
-            self.assertFailedRequest(self.run_request(), 'Valid e-mail address required.')
+            self.assertEqual(self.do_email_validation(email), 'Valid e-mail address required.')
 
     def test_change_email_to_existing_value(self):
         """ Test the error message if user attempts to change email to the existing value. """
-        self.request.POST['new_email'] = self.user.email
-        self.assertFailedRequest(self.run_request(), 'Old email is the same as the new email.')
-
-    def check_duplicate_email(self, email):
-        """Test that a request to change a users email to `email` fails"""
-        request = self.req_factory.post('unused_url', data={
-            'new_email': email,
-            'password': 'test',
-        })
-        request.user = self.user
-        self.assertFailedRequest(self.run_request(request), 'An account with this e-mail already exists.')
+        self.assertEqual(self.do_email_validation(self.user.email), 'Old email is the same as the new email.')
 
     def test_duplicate_email(self):
         UserFactory.create(email=self.new_email)
-        self.check_duplicate_email(self.new_email)
-
-    def test_capitalized_duplicate_email(self):
-        """Test that we check for email addresses in a case insensitive way"""
```

Comment:
Note to reviewer: this case insensitivity test was bogus-- the code actually IS case sensitive. It appeared to work because the request being passed in was ignored, and instead the test was using the email address assigned to a user in the setup method. Nothing about case sensitivity was actually being tested.